python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,483 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+import pytest
+
+import pyarrow as pa
+from pyarrow import compute as pc
+import random
+
+# UDFs are all tested with a dataset scan
+pytestmark = pytest.mark.dataset
+
+
+try:
+    import pyarrow.dataset as ds
+except ImportError:
+    ds = None
+
+
+@pytest.fixture(scope="session")
+def mock_udf_context():
+    # The batch_length is 1 for nullary functions.
+    # batch_length=1 since the udf expected output
+    # is generated by calling the udf function external
+    # to the call_function framework.
+    from pyarrow._compute import _get_scalar_udf_context
+    return _get_scalar_udf_context(pa.default_memory_pool(), 1)
```

Comment:
A length of `1` is wrong for most inputs, isn't it?

Reply:
Yes, you're correct. Added a higher default value (10) and customized depending on the usage.